### DISCUSSION TITLE
Transformer model (v3) doesn't do sentence segmentation

### DISCUSSION THREAD

**Discussion by Nuccy90** [explosion/spaCy#6987] [Help: Coding & Implementations] [answered]
Hi,

I was super excited when I saw that spaCy 3.0 had been released and I immediately started to train a transformer model for tagging and parsing (Swedish). When I tested the model, the tagger and parser work just fine but the model doesn't do sentence segmentation. The text comes out as one long sentence no matter what I feed into it.

I was looking around in old issues and I saw mentions of a pipline component that must be added in order to get sentence segmentation but I couldn't figure out whether this is still valid for the transformer-based models. I also tried to add the flag `-n 10` when I converted the old json data into .spacy format, but to no avail.

Any suggestions on what I can do about this?

**Comment by honnibal** [comment, marked as answer]
That does sound strange, I can't see why the parser isn't inserting the sentence boundaries. Have you sanity checked the binarised training data to see that it does indeed have multiple sentences per `Doc` object? You can load it in with the `DocBin` class like this:

```python
import spacy
from spacy.tokens import DocBin

nlp = spacy.blank("sv")
docbin = DocBin().from_disk(loc)
docs = list(docbin.get_docs(nlp.vocab))
```

The stages this could be going wrong at are:

1. There's only one sentence per `Doc` in the training data (so it never learns to predict `Break`)
2. The training data does have multiple sentences, but the parser somehow never learns them
3. The model does learn them during training, but somehow never predicts them at runtime

3) would definitely be a bug in spaCy. 2) might be a bug, but I suppose if the data is weird enough in some respect it might happen. 1) could be a bug in the conversion or preprocessing code, or just some sort of mistake.

**Reply by Nuccy90** [reply]
Thank you, I did the sanity check and the `-n 10` flag wasn't doing anything in the conversion from .json to .spacy. Now I retrained with paragraphs on 10 sentences and the sentence segmentation works fine!
